arch/aarch64/mcount-support.c

```diff
@@ -135,7 +169,12 @@ void mcount_get_stack_arg(struct mcount_arg_context *ctx,
 	if (offset < 1 || offset > 100)
 		pr_dbg("invalid stack offset: %d\n", offset);
 
-	memcpy(ctx->val.v, ctx->stack_base + offset, spec->size);
+	addr += offset;
+
+	if( check_arg_mem_region(ctx->regions, (unsigned long)addr) )
+		memcpy(ctx->val.v, addr, spec->size);
+	else
+		pr_err_ns("invalid argument: %d\n", offset);
```

Comment:
I don't think we want to terminate the program (`pr_err_ns()` will do) because of an invalid argument.  Probably `pr_dbg()` would be enough.  Or maybe we could zero-fill for such invalid arguments or offsets.